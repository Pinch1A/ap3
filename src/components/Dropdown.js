import React from 'react'
import DropdownListWrap from './DropdownListWrap'
import styled from 'styled-components'
import { ChevronDown } from '../assets/icons'

export const Flag = styled.img`
  width: 17px;
  height: 12px;
  margin-right: 10px;
`

const FlexWrap = styled.div`
  display: flex;
  flex-direction: column;
  background-color: #fff;
  width: 432px;
`
const DropdownHeader = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  border: 1px solid;
  border-color: ${(props) => (props.isOpen ? '#1300cb' : '#decec8')};
  padding: 10px 20px;
  border-radius: 6px;
  color: #777;
`
DropdownHeader.displayName = 'DropdownHeader'

const Input = styled.input`
  border: none;
  height: 20px;
  width: 100%;
  outline: none;
`

const Dropdown = ({
  headerTitle,
  data,
  isOpen,
  loadMore,
  hasMore,
  setOpen,
  toggleList,
  selected,
  setSelected,
  setSearchInput,
}) => {
  const handleOnChange = (val) => {
    setSelected({ name: '', flag: null })
    setSearchInput(val.target.value)
    setOpen(true)
  }

  const valueInput = selected.name !== '' ? selected.name : headerTitle

  return (
    <FlexWrap>
      <DropdownHeader tabIndex="0" isOpen={isOpen} onClick={() => toggleList()}>
        {selected.flag && <Flag src={selected.flag} alt={selected.name} />}
        <Input value={valueInput} placeholder="Select" type="text" onChange={handleOnChange} />
        <ChevronDown />
      </DropdownHeader>
      {isOpen && (
        <DropdownListWrap
          loadMore={loadMore}
          hasMore={hasMore}
          data={data}
          setSelected={setSelected}
          selected={selected}
          setOpen={setOpen}
        />
      )}
    </FlexWrap>
  )
}

export default Dropdown
