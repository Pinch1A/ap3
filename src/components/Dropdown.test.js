import Dropdown from './Dropdown'
import React from 'react'
import { shallow, mount } from 'enzyme'

const props = {
  headerTitle: 'Select',
  data: [],
  isOpen: false,
  loadMore: () => jest.fn(),
  hasMore: false,
  setOpen: () => jest.fn(),
  toggleList: () => jest.fn(),
  selected: { name: '', flag: null },
  setSelected: () => jest.fn(),
  setSearchInput: () => jest.fn(),
}

describe('<Dropdown>', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Dropdown {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Shows items list when header is clicked', () => {
    let isOpen = false
    const mockToggle = jest.fn(() => !isOpen)
    const props = {
      headerTitle: 'Select',
      data: [{ name: 'test', flag: 'test-flag' }],
      isOpen,
      loadMore: () => jest.fn(),
      hasMore: false,
      setOpen: () => jest.fn(),
      toggleList: mockToggle,
      selected: { name: '', flag: null },
      setSelected: () => jest.fn(),
      setSearchInput: () => jest.fn(),
    }

    const wrapper = mount(<Dropdown {...props} />)
    const dropdownHeader = wrapper.find('DropdownHeader')
    expect(wrapper.props().isOpen).toEqual(false)
    dropdownHeader.simulate('click')
    expect(props.toggleList.mock.calls.length).toEqual(1)
  })
})
