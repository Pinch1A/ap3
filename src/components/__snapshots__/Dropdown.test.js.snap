// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown> renders without crashing 1`] = `
<styled.div>
  <DropdownHeader
    isOpen={false}
    onClick={[Function]}
    tabIndex="0"
  >
    <styled.input
      onChange={[Function]}
      placeholder="Select"
      type="text"
      value="Select"
    />
    <ChevronDown />
  </DropdownHeader>
</styled.div>
`;
